.github/workflows/ci.yml

```
@@ -644,3 +673,121 @@ jobs:
         run: python3 ./qa/zcash/full_test_suite.py sec-hard
         env:
           HOST: ${{ matrix.host }}
+
+  rpc-depends:
+    name: RPC set up tier ${{ matrix.tier }} platform ${{ matrix.platform }}
+    needs: setup
+    runs-on: ${{ matrix.build_os }}
+    continue-on-error: ${{ matrix.tier != 1 }}
+    strategy:
+      matrix:
+        name: ${{ fromJson(needs.setup.outputs.rpc_test_names) }}
+        include: ${{ fromJson(needs.setup.outputs.rpc_test_matrix) }}
+
+    steps:
+      - name: Get Python dependencies for RPC tests
+        run: |
+          python3 -m venv ./venv
+          source ./venv/bin/activate
+          pip install simplejson zmq asyncio
+
+      - uses: actions/cache/save@v3
+        name: cache-venv
+        with:
+          enableCrossOsArchive: true
+          path: venv
+          key: venv-${{ matrix.name }}
+
+  test-rpc:
+    name: RPC tests tier ${{ matrix.tier }} platform ${{ matrix.platform }} test ${{ matrix.rpc_tests }}
+    needs: [setup, build, rpc-depends]
+    runs-on: ${{ matrix.build_os }}
```

This should also use `test_os` unless we specifically want to use faster paid runners; we may in fact want to use them for specific heavier RPC tests but not for the general RPC tests. If we want to do the latter, we'd introduce an `rpc_test_os` instead of reusing `build_os` (because we may want a different runner config to hit a different performance point for testing vs building). But for now:
```suggestion
    runs-on: ${{ matrix.test_os }}
```